### PR TITLE
[libclang/python] Fix bugs in custom enum implementation and add tests

### PR DESCRIPTION
**Pull request by DeinAlptraum**
Do not allow initialization of enum from negative IDs (e.g. from_id(-1) currently produces the last known variant)
Rename duplicate enums: CursorKind.OMP_TEAMS_DISTRIBUTE_DIRECTIVE and TypeKind.OBJCCLASS
Add tests to cover these cases

### CODE REVIEW

**Comment by DeinAlptraum** [comment]
@Endilll I separated the fixes for the enum bugs from the strict typing PR, and added tests that cover these cases. I checked that the tests fail before the fixes and succeed now.

File: clang/bindings/python/clang/cindex.py [hunk 1]
```diff
@@ -1336,7 +1336,7 @@ def __repr__(self):
 CursorKind.OMP_TEAMS_DISTRIBUTE_DIRECTIVE = CursorKind(271)
 
 # OpenMP teams distribute simd directive.
-CursorKind.OMP_TEAMS_DISTRIBUTE_DIRECTIVE = CursorKind(272)
+CursorKind.OMP_TEAMS_DISTRIBUTE_SIMD_DIRECTIVE = CursorKind(272)
```
Comment (DeinAlptraum) [inline clang/bindings/python/clang/cindex.py:1339]: This was a duplicate with variant 271. new name taken from https://clang.llvm.org/doxygen/Index_8h_source.html, line 2011

File: clang/bindings/python/clang/cindex.py [hunk 2]
```diff
@@ -2395,7 +2395,7 @@ def __repr__(self):
 TypeKind.OCLRESERVEID = TypeKind(160)
 
 TypeKind.OBJCOBJECT = TypeKind(161)
-TypeKind.OBJCCLASS = TypeKind(162)
+TypeKind.OBJCTYPEPARAM = TypeKind(162)
```
[the remark this answers is not in the document]
Comment (Endilll) [reply clang/bindings/python/clang/cindex.py:2398]: Same as for OpenMP enum.

File: clang/bindings/python/clang/cindex.py (hunk 1 above)
Comment (Endilll) [reply clang/bindings/python/clang/cindex.py:1339]: LGTM, but worth an entry in release notes under "potentially breaking changes".

File: clang/bindings/python/tests/cindex/test_enums.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        """Check that no kind name has been used multiple times"""
+        for enum in self.enums:
+            seen_names = set()
+            for id in range(len(enum._kinds)):
+                try:
+                    kind_name = enum.from_id(id).name
+                except ValueError:
+                    continue
+                self.assertNotIn(kind_name, seen_names)
+                seen_names.add(id)
```
Comment (Endilll) [inline clang/bindings/python/tests/cindex/test_enums.py:50]: I think you can just add them to the set, and then check whether it has the same length as `enum._kinds`.

Comment (DeinAlptraum) [reply clang/bindings/python/tests/cindex/test_enums.py]: I only just realized that even that is not necessary, calling `enum.from_id(id).name` already fails for duplicate IDs, so no assertions needed.

**Review by Endilll — APPROVED** [review]

**Comment by Endilll** [comment]
Let me know when you think this is good to go.

**Comment by DeinAlptraum** [comment]
These were all the bugs I'd found, so I think it is

**Comment by Endilll** [comment]
This depends on #95210 to pass the CI. Let's merge that first.

**Pull request merged**